packages/core/src/urlPolyfill.ts

```diff
@@ -15,17 +15,34 @@ export function haveSameOrigin(url1: string, url2: string) {
 }
 
 export function getOrigin(url: string) {
-  return new URL(url).origin
+  return buildUrl(url).origin
 }
 
 export function getPathName(url: string) {
-  return new URL(url).pathname
+  return buildUrl(url).pathname
 }
 
 export function getSearch(url: string) {
-  return new URL(url).search
+  return buildUrl(url).search
 }
 
 export function getHash(url: string) {
-  return new URL(url).hash
+  return buildUrl(url).hash
+}
+
+function buildUrl(url: string, base?: string) {
```

Comment:
I think `buildUrl` should use `URL` if available (it would improve performances)